src/lib/scan-imports.ts

```diff
@@ -0,0 +1,217 @@
+        importModuleName(node as NodePath<ImportDeclaration>) ||
+        dynamicImportModuleName(node as NodePath<CallExpression>);
+
+      if (!moduleName) {
+        return;
+      }
+
+      // if remote module
+      if (moduleName.startsWith('http://') || moduleName.startsWith('https://')) {
+        if (!deps.includes(moduleName)) {
+          deps.push(moduleName);
+        }
+        return;
+      }
+
+      try {
+        // if local module
+        const resolvedPath = path.resolve(path.dirname(file), moduleName);
+        if (exists(resolvedPath) && !deps.includes(resolvedPath)) {
+          deps.push(resolvedPath.replace(cwd, '')); // strip out full path above project root (cwd) when adding
+        }
+      } catch (e) {
+        // if npm module
+        if (!deps.includes(moduleName)) {
+          deps.push(moduleName);
+        }
+      }
+    },
+  });
+
+  return [...deps].sort((a, b) => a.localeCompare(b));
+}
+
+/**
+ * Check if filepath is valid
+ */
+
+function exists(filename: string) {
+  const stats = fs.statSync(filename);
+  return stats && stats.isFile();
+}
+
+/**
+ * Take an array of files, recursively scan imports, and return dependencies
+ */
+function buildDependencyTree(
+  files: string[],
+  dependencyTree: DependencyTree = {},
+  npmDependencies: string[] = [],
+) {
+  files.forEach(file => {
+    const key = file.replace(cwd, ''); // don’t expose full filepath
+
+    // if file already parsed, skip
+    if (dependencyTree[key]) {
+      return;
+    }
+
+    // if new module, add to tree
+    const modules = getDependencies(file);
+    dependencyTree[key] = modules;
+
+    // resolve modules
+    const localModules: string[] = [];
+    modules.forEach(moduleName => {
+      if (moduleName.startsWith('http://') || moduleName.startsWith('https://')) {
+        return; // skip remote modules
+      }
+
+      // if local, try and resolve (this will throw if missing)
+      try {
+        const localModule = path.resolve(cwd, moduleName.replace(/^\//, ''));
+        if (exists(localModule)) {
+          localModules.push(localModule);
+        }
+      } catch (e) {
+        // if file can’t be located, it might be an npm module
+        if (!npmDependencies.includes(moduleName)) {
+          npmDependencies.push(moduleName);
+        }
+      }
+    });
+
+    // recursively scan these modules until there are none left
+    if (localModules.length) {
+      buildDependencyTree(localModules, dependencyTree, npmDependencies);
+    }
+  });
+
+  return {dependencyTree, npmDependencies};
+}
+
+interface ScanOptions {
+  dest: string;
+}
+
+interface ScanResults {
+  dependencies: string[];
+  dependencyTree?: DependencyTree;
+}
+
+/**
+ * Scan files and their imports for dependencies
+ */
+export default function scanImports(entry: string, options: ScanOptions): ScanResults {
+  // return on missing entry
+  if (!entry.length) {
+    return {dependencies: []};
+  }
+
+  const files = glob.sync(entry);
```

Comment:
I added glob, and though there’s the potential to scan/resolve imports the user isn’t using, I didn’t find it added significant time in parsing.